File: 2015/AI/AIClient_Python/src/event/AddPlayerEvent.py
'''
Created on Dec 17, 2014

@author: scarriere
'''
from event.OutgoingEvent import OutgoingEvent

class AddPlayerEvent(OutgoingEvent):

    def __init__(self, teamName, characterNames):
        self.teamName = teamName
        self.characterNames = characterNames
        
    def toString(self):
        message = "Game:AddPlayer:"
        message += self.teamName
        
        for characterName in self.characterNames:
            message += self.SEPARATOR + characterName
        
        return message

File: 2015/AI/AIClient_JAVA/src/event/AddPlayerEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** AddPlayerEvent.java
 ** Event call to join game
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

public class AddPlayerEvent extends OutgoingEvent {
	private String teamName;
	private String[] characterNames;

	public AddPlayerEvent(String teamName, String[] characterNames) {
		this.teamName = teamName;
		this.characterNames = characterNames;
	}

	@Override
	public String toString() {
		String message = "Game:AddPlayer" + SEPARATOR;
		message += teamName;

		for (int index = 0; index < characterNames.length; ++index) {
			message += SEPARATOR + characterNames[index];
		}
		return message;
	}
}
